test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+            });
+        });
+    });
+
+    describe("#vaultsCount", () => {
+        it("returns the number of registered vaults", async () => {
+            expect(await this.subject.vaultsCount()).to.be.equal(2);
+        });
+
+        describe("access control:", () => {
+            it("allowed: any address", async () => {
+                await withSigner(randomAddress(), async (s) => {
+                    await expect(this.subject.vaultsCount()).to.not.be.reverted;
+                });
+            });
+        });
+        describe("edge cases", () => {
+            describe("when new vault is registered", () => {
+                it("is increased by 1", async () => {
+                    let oldVaultsCount = Number(
+                        await this.subject.vaultsCount()
+                    );
+                    await withSigner(
+                        this.erc20VaultGovernance.address,
+                        async (s) => {
+                            await this.subject
+                                .connect(s)
+                                .registerVault(
+                                    this.anotherERC20Vault.address,
+                                    await this.ownerSigner.getAddress()
+                                );
+                        }
+                    );
+                    this.nftAnotherERC20 = Number(
+                        await this.subject.vaultsCount()
+                    );
+                    let newVaultsCount = Number(
+                        await this.subject.vaultsCount()
+                    );
+                    expect(newVaultsCount - oldVaultsCount).to.be.equal(1);
+                });
+            });
+        });
+    });
+
+    describe("#registerVault", () => {
+        it("binds minted ERC721 NFT to Vault address and transfers minted NFT to owner specified in args", async () => {
+            let newOwner = randomAddress();
+            expect(await this.subject.balanceOf(newOwner)).to.be.equal(0);
+            await withSigner(this.erc20VaultGovernance.address, async (s) => {
+                await this.subject
+                    .connect(s)
+                    .registerVault(this.anotherERC20Vault.address, newOwner);
+            });
+            expect(await this.subject.balanceOf(newOwner)).to.be.equal(1);
+        });
+        it("emits VaultRegistered event", async () => {
+            await withSigner(this.erc20VaultGovernance.address, async (s) => {
+                await expect(
+                    this.subject
+                        .connect(s)
+                        .registerVault(
+                            this.anotherERC20Vault.address,
+                            this.ownerSigner.address
+                        )
+                ).to.emit(this.subject, "VaultRegistered");
+            });
+        });
+
+        describe("properties", () => {
+            it("@property: minted NFT equals to vaultRegistry#vaultsCount", async () => {
```

Comment:
`it("@property: ...)` is a symbolic name. The actual test should use `pit("...")` function. Check Vault Governances tests